FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c

```diff
@@ -72,4 +76,1421 @@ int suiteTearDown( int numFailures )
     return numFailures;
 }
 
+/* =============================  HELPER FUNCTIONS  ========================= */
+void vApplicationMinimalIdleHook( void )
+{
+    /* Adding this function to pass compilation. */
+}
+
 /* ==============================  Test Cases  ============================== */
+
+/**
+ * @brief AWS_IoT-FreeRTOS_SMP_TC-89
+ * A task of equal priority will be created for each available CPU core. An
+ * additional task will be created in the ready state. This test will verify
+ * that as OS ticks are generated the ready task will be made to run on each
+ * CPU core.
+ *
+ * #define configRUN_MULTIPLE_PRIORITIES                    1
+ * #define configUSE_TIME_SLICING                           1
+ * #define configUSE_CORE_AFFINITY                          1
+ * #define configNUMBER_OF_CORES                            (N > 1)
+ *
+ * This test can be run with FreeRTOS configured for any number of cores greater than 1 .
+ *
+ * Tasks are created prior to starting the scheduler.
+ *
+ * Task (TN)	    Task (TN + 1)
+ * Priority – 1     Priority – 1
+ * State - Ready	State - Ready
+ *
+ * After calling vTaskStartScheduler()
+ *
+ * Task (TN)	               Task (TN + 1)
+ * Priority – 1                Priority – 1
+ * State - Running (Core N)	   State - Ready
+ *
+ * Call xTaskIncrementTick() for each configured CPU core.
+ *
+ * Task (TN + 1) when configNUMBER_OF_CORES = 4
+ * Tick    Core
+ * 1       0
+ * 2       1
+ * 3       2
+ * 4       3
+ */
+void test_timeslice_verification_tasks_equal_priority( void )
+{
+    TaskHandle_t xTaskHandles[ configNUMBER_OF_CORES + 1 ] = { NULL };
+    uint32_t i;
+
+    /* Create configNUMBER_OF_CORES + 1 tasks of equal priority */
+    for( i = 0; i < ( configNUMBER_OF_CORES + 1 ); i++ )
+    {
+        xTaskCreate( vSmpTestTask, "SMP Task", configMINIMAL_STACK_SIZE, NULL, 1, &xTaskHandles[ i ] );
+    }
+
+    vTaskStartScheduler();
+
+    /* Verify all configNUMBER_OF_CORES tasks are in the running state */
+    for( i = 0; i < configNUMBER_OF_CORES; i++ )
+    {
+        verifySmpTask( &xTaskHandles[ i ], eRunning, i );
+    }
+
+    /* Verify the last task is in the ready state */
+    verifySmpTask( &xTaskHandles[ i ], eReady, -1 );
+
+    /* Generate a tick for each configNUMBER_OF_CORES. This will cause each
+     * task to be either moved to the ready state or the running state */
+    for( i = 0; i < configNUMBER_OF_CORES; i++ )
+    {
+        xTaskIncrementTick_helper();
+
+        /* Verify the last created task runs on each core or enters the ready state */
+        verifySmpTask( &xTaskHandles[ configNUMBER_OF_CORES ], eRunning, i );
```

Comment:
The comment is incorrect, you never verify the task enters the ready state.
Do you need to increment the tick one more time so the task enters the ready state?